### ISSUE TITLE
Question: (Feature Request) update existing NavItemView

### ISSUE THREAD

**Issue by Alex-Chopard** [InboxSDK/InboxSDK#870]
**Question(s)**
I've always wondered why the class `NavItemView` don't have a method to update its content (icon/name/subtitle), it's wanted or it's just something not developed? And, it is something you will be ready to implement?

I've made a POC to test if it was something easily implementable:
branch: https://github.com/Alex-Chopard/InboxSDK/tree/pocNavItemUpdate
diff: https://github.com/Alex-Chopard/InboxSDK/compare/main...pocNavItemUpdate

**Comment by Macil** [comment]
You can update the content if you pass sdk.NavMenu.addNavItem() a stream/observable created with the RxJS or Kefir libraries. We have some documentation about this at https://inboxsdk.github.io/inboxsdk-docs/resources/#streams.

It was decided early on in the InboxSDK's design to accept stream objects like these as parameters as a universal way to accept values that may change over time from the user, and to encourage patterns that allowed the InboxSDK to signal disinterest and unsubscribe from changing values in a standard way to help prevent leaks in applications using the InboxSDK. Over time there have been places where the principle didn't fit well and hasn't been followed (like [MoleView](https://github.com/InboxSDK/InboxSDK/blob/main/src/platform-implementation-js/widgets/mole-view.js), which isn't created from a stream, has regular OOP setters and emits a "destroy" event when the user should stop trying to update it; it would be very awkward to control its minimization through a stream instead of a setter method) and I think we're less wed to it than we used to be, but we're not sure if we want to add redundant updating APIs or deprecate existing stream support in some APIs.

If we were to add an update method to NavItemView making it a kind of hybrid API, maybe the ideal would be to keep the hybrid nature of it as limited as possible in the uppermost layer of the API and only use a stream or OOP-style API for updates with the underlying GmailNavItemView, but maybe strictly following that is more effort than it's worth here. Your branch may be a good base, though I'd prefer that updateNavItem didn't return a Promise so to not expose the underlying asynchrony in the API, that it didn't accept a new stream (smallest change would be to swap out `kefirCast()` for `Kefir.constant()`), and that `GmailNavItemView.setNavItemDescriptor()`, which was previously made to only be used once, was changed to unsubscribe from a previous stream, for the case it was initialized with one:

<details>
  <summary>setNavItemDescriptor repeatable call patch</summary>

```diff
diff --git a/src/platform-implementation-js/dom-driver/gmail/views/gmail-nav-item-view.ts b/src/platform-implementation-js/dom-driver/gmail/views/gmail-nav-item-view.ts
--- src/platform-implementation-js/dom-driver/gmail/views/gmail-nav-item-view.ts
+++ src/platform-implementation-js/dom-driver/gmail/views/gmail-nav-item-view.ts
@@ -75,8 +75,9 @@
   private _accessoryViewController: any = null;
   private _driver: GmailDriver;
   private _element: HTMLElement;
   private _eventStream: Bus<any, any>;
+  private _resetNavItemDescriptor = kefirBus<null, never>();
   private _expandoElement: HTMLElement | null = null;
   private _iconSettings: IconSettings = {};
   private _isActive: boolean = false;
   private _isCollapsed: boolean = false;
@@ -136,8 +137,9 @@
     return gmailNavItemView;
   }
 
   destroy() {
+    this._resetNavItemDescriptor.end();
     this._element.remove();
     if (this._accessoryViewController) this._accessoryViewController.destroy();
     if (this._eventStream) this._eventStream.end();
     if (this._expandoElement) this._expandoElement.remove();
@@ -229,9 +231,11 @@
 
   setNavItemDescriptor(
     navItemDescriptorPropertyStream: Kefir.Observable<any, any>
   ) {
+    this._resetNavItemDescriptor.value(null);
     navItemDescriptorPropertyStream
+      .takeUntilBy(this._resetNavItemDescriptor)
       .takeUntilBy(this._eventStream.filter(() => false).beforeEnd(() => null))
       .onValue((x) => this._updateValues(x));
   }
 ```

</details>

**Comment by Alex-Chopard** [comment]
I've tested the stream with Kefir to update a NavItemView, and it fits totally with my needs 🙏.

The only issue I've met is with the update of the "subtitle", the update appends the new value in the DOM instead of replacing it.
https://github.com/InboxSDK/InboxSDK/blob/7a2906d79bdfad8ec5f28e1479ba3c4d70d63b21/src/platform-implementation-js/dom-driver/gmail/views/gmail-nav-item-view.ts#L970

**Comment by wegry** [comment]
Should be fixed in #879. Thanks @Alex-Chopard.
